Untitled-Engine/Src/Untitled/Core/Events/KeyEvent.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 [Untitled Games]
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE', which is part of this source code package.
+// 
+// [+ :]
+
+#pragma once
+
+#include "Event.h"
+
+namespace ue
+{
+
+	class KeyEvent : public Event
+	{
+	public:
+		KeyEvent(const int keycode) : m_KeyCode(keycode) {}
+		KeyEvent() = delete;
+		_NODISCARD inline int GetKeyCode() const { return m_KeyCode; }
+		EVENT_CLASS_CATEGORY(int(EventCategory::EventCategoryKeyboard) | int(EventCategory::EventCategoryInput))
```

Comment:
When making bitfield here, maybe we should consider having event categories as defines instead of enum.

That would take away the need to cast and in my opinion, make this line more readable

eg.
UE_EVENTCAT_KEYBOARD | UE_EVENTCAT_INPUT

Reply:
Macros do not obey scope and type rules. Also, macro names are removed during preprocessing and so usually don’t appear in tools like debuggers. - cpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-macro

Wouldn't it be better to overload the | operator to work with EventCategory enum types, which removes the need for the cast too?